### ISSUE TITLE
Debug curl calls during a build

### ISSUE THREAD

**Issue by jshort** [NixOS/nixpkgs#314364]
## Issue description
I'm working to get nix-build to work in a corporate environment through a proxy.  It seems to be able to make http/curl GETs for things in some cases and in other cases not and I get:

`curl: (5) Could not resolve proxy: myproxy`

I'm setting the env vars `http_proxy`, `https_proxy`, and `ftp_proxy` though I have confirmed through debugging it is only looking at `https_proxy`.

The following command works fine:

`https_proxy=http://myproxy:8080 /usr/bin/curl https://sources.debian.org/data/main/s/sharutils/1:4.15.2-2+deb9u1/debian/patches/01-fix-heap-buffer-overflow-cve-2018-1000097.patch`

But for some reason when nix-build runs the package build, it can't resolve `myproxy` from `/etc/hosts`.

Is there a way to debug `nix-build` to see all the commands being invoked for `fetchurl` or `fetchpatch`?

**Comment by jshort** [comment]
Oh yuck:

```
https_proxy=myproxy:8080 /usr/bin/curl -4 https://sources.debian.org/data/main/s/sharutils/1:4.15.2-2+deb9u1/debian/patches/01-fix-heap-buffer-overflow-cve-2018-1000097.patch
curl: (5) Could not resolve proxy: fwdproxy
```
So it seems that `nix-build` is passing `-4` to force ipv4.  Is there a way to override this?

**Comment by jshort** [comment]
Issue was this: https://github.com/NixOS/nixpkgs/pull/277471
